dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/relationship/hibernate/HibernateRelationshipStore.java

```diff
@@ -311,6 +311,19 @@ public List<Relationship> getByUids( List<String> uids )
         }
     }
 
+    @Override
+    protected void preProcessPredicates( CriteriaBuilder builder,
+        List<Function<Root<Relationship>, Predicate>> predicates )
+    {
+        predicates.add( root -> builder.equal( root.get( "deleted" ), false ) );
+    }
+
+    @Override
+    protected Relationship postProcessObject( Relationship relationship )
+    {
+        return (relationship == null || relationship.isDeleted()) ? null : relationship;
```

Comment:
I assume this removes deleted rels before they are returned from the store methods?

Reply:
Yes, this is used by get method.
I am really not sure why here is not used the preProcessPredicates as well but this is not part of this PR